### ISSUE TITLE
Qualified name resolution not working in rule tests

### ISSUE THREAD

**Issue by bibz** [sourcery-ai/sourcery#275]
### Checklist

- [X] I have searched the [Sourcery documentation](https://docs.sourcery.ai/) for the issue, and found nothing
- [X] I have checked there are no open [bugs](https://github.com/sourcery-ai/sourcery/issues?q=label%3Abug) referencing the same bug or problem

### Description

According to the pattern syntax documentation, [qualified name resolution](https://docs.sourcery.ai/custom_rules/reference/pattern-syntax/#qualified-name-resolution) should work out of the box.

Defining a sourcery configuration with the example rule from the documentation is not working, the IDE plugin (and the CLI as well) complains the tests do not pass.

```yaml
# .sourcery.yaml
rules:
  - id: avoid-logging-debug
    description: Remove logging.debug calls from code
    pattern: logging.debug
    tests:
      # literal matches still happen, so you do not need to import `logging`
      # here:
      - match: logging.debug(DEBUG_MESSAGE)
      - no-match: logging.info(INFO_MESSAGE)
      # NOTE: the following test matches because of name resolution.
      # The name `lg` refers to `logging`, so `lg.debug` resolves to
      # `logging.debug`.
      - match: |
          import logging as lg

          lg.debug(DEBUG_MESSAGE)
      # NOTE: the following test matches because of name resolution.
      # The name `dbug` refers to `logging.debug` because of the aliased
      # import.
      - match: |
          import logging.debug as dbug

          dbug(DEBUG_MESSAGE)
      # NOTE: the following test matches because of name resolution.
      # The name `debug` refers to `logging.debug` because it was directly
      # imported from the module `logging`.
      - match: |
          from logging import debug

          debug(DEBUG_MESSAGE)
```

```sh
% sourcery review
Error loading config:
.sourcery.yaml:12:8-12:13 (rules -> 0 -> tests -> 2 -> match) Match example did not trigger
.sourcery.yaml:26:8-26:13 (rules -> 0 -> tests -> 4 -> match) Match example did not trigger
```
I would expect that the example rule from the documentation is supported by the tool.

### Debug Information

IDE Version: `PyCharm 2022.2.1 (Professional Edition)`
Sourcery Version: `sourcery, version 0.12.7`

Operating system and Version: Debian 11.4 (`Linux 5.10.0-17-amd64 #1 SMP Debian 5.10.136-1 (2022-08-13) x86_64 GNU/Linux`)

**Comment by brendanator** [comment]
Thanks for reporting this issue @bibz 

I have literally just discovered this issue internally myself!

I have a fix for it which will be in the next release of Sourcery

**Comment by brendanator** [comment]
Note that this is only failing in the config. If you comment out those failing tests then it will work in the rest of your codebase

**Comment by bibz** [comment]
> Note that this is only failing in the config. If you comment out those failing tests then it will work in the rest of your codebase

Ah, good to know!
Indeed, as you mention, without the test my rule works :+1:

**Comment by Hellebore** [comment]
Resolved in 0.12.8
